DOCUMENTATION.md

```diff
@@ -3090,317 +2814,689 @@ Client.newHazelcastClient().then(function (client) {
-}).then(function () {
-    // now create the orders for this customer
-    return mapOrders.putAll([
-        [new OrderKey(21, 1), order],
-        [new OrderKey(22, 1), order],
-        [new OrderKey(23, 1), order]
-    ]);
-});
-```
+    log: function (level, objectName, message, furtherInfo) {
+        this.logger.log(this.levels[level], objectName + ': ' + message, furtherInfo);
+    },
 
-For more details, see the [PartitionAware section](https://docs.hazelcast.org/docs/latest/manual/html-single/#partitionaware) in the Hazelcast IMDG Reference Manual.
+    error: function (objectName, message, furtherInfo) {
+        this.log(LogLevel.ERROR, objectName, message, furtherInfo);
+    },
 
-### 7.8.2. Near Cache
+    warn: function (objectName, message, furtherInfo) {
+        this.log(LogLevel.WARN, objectName, message, furtherInfo);
+    },
 
-Map entries in Hazelcast are partitioned across the cluster members. Hazelcast clients do not have local data at all. Suppose you read the key `k` a number of times from a Hazelcast client and `k` is owned by a member in your cluster. Then each `map.get(k)` will be a remote operation, which creates a lot of network trips. If you have a map that is mostly read, then you should consider creating a local Near Cache, so that reads are sped up and less network traffic is created.
+    info: function (objectName, message, furtherInfo) {
+        this.log(LogLevel.INFO, objectName, message, furtherInfo);
+    },
 
-These benefits do not come for free, please consider the following trade-offs:
+    debug: function (objectName, message, furtherInfo) {
+        this.log(LogLevel.DEBUG, objectName, message, furtherInfo);
+    },
 
-- If invalidation is enabled and entries are updated frequently, then invalidations will be costly.
+    trace: function (objectName, message, furtherInfo) {
+        this.log(LogLevel.TRACE, objectName, message, furtherInfo);
+    }
 
-- Near Cache breaks the strong consistency guarantees; you might be reading stale data.
+};
+cfg.customLogger = winstonAdapter;
+```
 
-- Clients with a Near Cache will have to hold the extra cached data, which increases memory consumption.
+Note that it is not possible to configure custom logging via declarative configuration.
 
-Near Cache is highly recommended for maps that are mostly read.
+## 7.10. Defining Client Labels
 
-#### 7.8.2.1. Configuring Near Cache
+Through the client labels, you can assign special roles for your clients and use these roles to perform some actions
+specific to those client connections.
 
-The following snippets show how a Near Cache is configured in the Node.js client, presenting all available values for each element:
+You can also group your clients using the client labels. These client groups can be blacklisted in the
+Hazelcast Management Center so that they can be prevented from connecting to a cluster.
+See the related section in the Hazelcast Management Center Reference Manual for more information on this topic.
```

Comment:
nit: this link could be placed somewhere here: https://docs.hazelcast.org/docs/management-center/latest/manual/html/index.html#changing-cluster-client-filtering